service/database/index.js

```diff
@@ -0,0 +1,175 @@
+const { getConnection } = require("./connector");
+const { transformIdToObjectId } = require("./util");
+
+/**
+ * @class Mongo
+ * @description Handles database operations, called via handler. This is like a generic that
+ * is used through the project to perform basic operations on the database.
+ */
+class Mongo {
+  /**
+   * Runs the MongoDB find() method to fetch documents.
+   *
+   * @async
+   * @param {string} database Name of the database
+   * @param {string} collectionName Name of the collection to run operation on
+   * @param {document} query Specifies selection filter using query operators.
+   * To return all documents in a collection, omit this parameter or pass an empty document ({}).
+   * @param {boolean} [transform=false] check to transform the IDs to ObjectID in response
+   *
+   * {@link https://docs.mongodb.com/manual/reference/method/db.collection.find/ Read MongoDB Reference}
+   */
+  static async find(database, collectionName, query, transform = true) {
+    try {
+      query = transformIdToObjectId(query);
+
+      const collection = getConnection(database).collection(collectionName);
+      const data = await collection.find(query).toArray();
+
+      /** allow caller method to toggle response transformation */
+      if (transform) {
+        data.forEach((x) => {
+          x["_id"] = {
+            $oid: x["_id"],
+          };
+        });
+      }
+
+      return data;
+    } catch (e) {
+      console.error(e);
+      return e;
```

Comment:
Does this error return trigger setting the response code appropriately?

Reply:
thanks for pointing this !

I isolated two functions to test the behavior, looks like the return statement has to be replaced with a throw statement. Below is a simple gist to demonstrate the behavior.

https://gist.github.com/232fa53a05d96fb24a925e35ac0e3db2

pushed commits for this change.